### ISSUE TITLE
Does ucx support allocate and register cuda malloc?

### ISSUE THREAD

**Issue by Keepmoving-ZXY** [openucx/ucx#4392]
These days I want to find out how to allocate cuda memory in ucx. In `libperf.c`, I find ucx use `cudaMalloc` as cuda memory allocator,  and no rcache to manage cuda memory as below:
```
static ucs_status_t ucp_perf_cuda_alloc(ucx_perf_context_t *perf, size_t length,
                                        void **address_p, ucp_mem_h *memh_p,
                                        int non_blk_flag)
{
    cudaError_t cerr;

    cerr = cudaMalloc(address_p, length);
    if (cerr != cudaSuccess) {
        return UCS_ERR_NO_MEMORY;
    }

    return UCS_OK;
}
```
In my project, I want to allocate cuda memory then register memory to RDMA NIC with `rdma_reg_read` as well as host memory. And these days I have a look at `ucp_mem_map`, it seems that  `rcache` has already manage host memory and can register memory to RDMA NIC if possible.  
 So does ucx's `ucp_mem_map` support allocate cuda memory and register it to RDMA NIC to enable GPUDriectRDMA?

**Comment by yosefe** [comment]
Yes, ucp_mem_map() can register cuda memory and manage it in rcache just like host memory

**Comment by Keepmoving-ZXY** [comment]
ok, I will have a more careful look at parameter of `ucp_mem_map`  to find out how to allocate cuda memory with `ucp_mem_map`.

**Comment by yosefe** [comment]
ucp_mem_map() can **register** cuda memory. it can't allocate cuda memory

**Comment by Keepmoving-ZXY** [comment]
thank you, let me have a try.
